src/main/java/hvac/roomcoordinator/RoomContext.java

```diff
@@ -0,0 +1,119 @@
+package hvac.roomcoordinator;
+
+import hvac.meetingontology.Meeting;
+import hvac.simulation.rooms.RoomWall;
+import jade.core.AID;
+
+import java.util.*;
+
+public class RoomContext {
+    private static final float defaultTemperature = 21;
+    private AID coordinator;
+    private AID myRoomUpkeeper;
+    private HashMap<AID, RoomWall> myNeighbours;
+    private Meeting currentMeeting;
+    private PriorityQueue<Meeting> meetingsQueue = new PriorityQueue<>();
+    private Map<String, AbstractMap.SimpleEntry<Meeting, Map<AID, Float>>> neighboursForecastStatus = new HashMap<>();
+
+    RoomContext(AID myRoomUpkeeper, AID coordinator, HashMap<AID, RoomWall>  myNeighbours){
+        this.coordinator = coordinator;
+        this.myRoomUpkeeper = myRoomUpkeeper;
+        this.myNeighbours = myNeighbours;
+    }
+
+    public AID getCoordinator() {
+        return coordinator;
+    }
+
+    @SuppressWarnings("unused")
+    public AID getMyRoomUpkeeper() {
+        return myRoomUpkeeper;
+    }
+
+    public HashMap<AID, RoomWall>  getMyNeighbours() {
+        return myNeighbours;
+    }
+
+    public Meeting getCurrentMeeting() {
+        return currentMeeting;
+    }
+
+    public void setCurrentMeeting(Meeting currentMeeting) {
+        this.currentMeeting = currentMeeting;
+    }
+
+    public void addMeeting(Meeting meeting){
+        meetingsQueue.add(meeting);
+    }
+
+    public Meeting peekMeeting(){
+        return meetingsQueue.peek();
+    }
+
+    public Meeting checkMeetings(Date date){
+        if (meetingsQueue.isEmpty() || 0 < meetingsQueue.peek().getStartDate().compareTo(date)){
+            return null;
+        }
+        return meetingsQueue.peek();
+    }
+
+    public boolean isTimeSlotAvailable(Meeting meeting){
+        return meetingsQueue.stream().noneMatch(meeting::isOverlapping);
+    }
+
+    public float estimatedTemperature(Meeting meeting){
```

Comment:
Wydaje mi się, że tu warto nazwać argument neighboursMeeting czy coś w tym stylu, bo bez kontekstu ta metoda trochę niejasna jest.

Reply:
chodzi o estimatedTemperature, żeby dać jasno do zrozumienia, że to jest tylko uproszczone obliczanie temperatury dla informacji sąsiadów, nie głównego koordynatora?